server/task/src/test/java/org/apache/james/task/TaskManagerContract.java

```diff
@@ -431,7 +434,41 @@ default void awaitShouldAwaitWaitingTask() {
         TaskId task2 = taskManager.submit(
             new CompletedTask());
 
-        assertThat(taskManager.await(task2).getStatus()).isEqualTo(TaskManager.Status.COMPLETED);
+        assertThat(taskManager.await(task2, TIMEOUT).unwrap().getStatus()).isEqualTo(TaskManager.Status.COMPLETED);
+    }
+
+    @Test
+    default void awaitANonExistingTaskShouldReturnAnUnknownAwaitedTaskExecutionDetailsAndThrow() {
+        TaskManager taskManager = taskManager();
+        TaskManager.AwaitedTaskExecutionDetails result = taskManager.await(TaskId.generateTaskId(), TIMEOUT);
+
+        assertThat(result).isInstanceOf(TaskManager.UnknownAwaitedTaskExecutionDetails.class);
+
+        assertThatCode(() -> result
+                .onTimeout(CustomTimeoutException::new)
+                .onUnknown(CustomUnknownException::new)
```

Comment:
Why is it the responsability of the caller to instanciate the exceptions?

Why can't we have well defined exceptions thrown directly by the `await` method?

Sorry I don't get the benefits behind all of this complexity...

Reply:
> Why is it the responsability of the caller to instanciate the exceptions?

I do now instantiate it here, I pass the method reference

> Why can't we have well defined exceptions thrown directly by the await method?

Because reaching a timeout is not a real exception, consequently, it should not be manage like that

> Sorry I don't get the benefits behind all of this complexity...

ADT are not complex, `RunTimeException`s are

Reply:
> I do now instantiate it here, I pass the method reference

Thanks I understand

> Because reaching a timeout is not a real exception, consequently, it should not be manage like that

Thanks that's a beginning of answer

> ADT are not complex, RunTimeExceptions are

Who said that it should be runtime?

Reply:
> Who said that it should be runtime?

Well, if it's not passed as a value, the only other non too-dirty way I see is an `Exception`

Reply:
That would be more straightforward, no?